plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/cleaner/DockerContainerCleaner.java

```diff
@@ -0,0 +1,123 @@
+                                  DockerConnector dockerConnector,
+                                  DockerContainerNameGenerator nameGenerator) {
+        this.machineManager = machineManager;
+        this.dockerConnector = dockerConnector;
+        this.nameGenerator = nameGenerator;
+    }
+
+    @ScheduleRate(periodParameterName = "machine.docker.unused_containers_cleanup_period_min",
+                  initialDelayParameterName = "machine.docker.unused_containers_cleanup_period_min",
+                  unit = TimeUnit.MINUTES)
+    @Override
+    public void run() {
+        try {
+            List<ContainerListEntry> allDockerContainers = dockerConnector.listContainers();
+            List<MachineImpl> machines = machineManager.getMachines();
+
+            List<ContainerListEntry> unUsedContainers = findUnUsedContainers(allDockerContainers, machines);
+
+            for (ContainerListEntry container : unUsedContainers) {
+                killContainer(container);
+                removeContainer(container);
+            }
+        } catch (IOException e) {
+            LOG.error("Failed to get list docker containers", e);
+        } catch (MachineException e) {
+            LOG.error("Failed to get list machines to clean up unused containers", e);
+        } catch (Exception e) {
+            LOG.error("Failed to clean up inactive containers", e);
+        }
+    }
+
+    private List<ContainerListEntry> findUnUsedContainers(List<ContainerListEntry> containers, List<MachineImpl> machines) {
+        List<ContainerListEntry> unUsedContainers = new ArrayList<>();
+        for (ContainerListEntry container : containers) {
+            Optional<ContainerNameInfo> optional = nameGenerator.parse(container.getNames()[0]);
+            if (!optional.isPresent()) {
+                continue;
+            }
+            final ContainerNameInfo containerNameInfo = optional.get();
+            boolean containerIsUsed = machines.stream()
+                                              .anyMatch(machine -> machine.getId().equals(containerNameInfo.getMachineId())
+                                                                   && machine.getWorkspaceId().equals(containerNameInfo.getWorkspaceId()));
+            if (!containerIsUsed) {
+                unUsedContainers.add(container);
+            }
+        }
+        return unUsedContainers;
+    }
+
+    private void killContainer(ContainerListEntry container) {
+        String containerId = container.getId();
+        String containerName = container.getNames()[0];
+        try {
+            if (container.getStatus().startsWith("Up")) {
+                dockerConnector.killContainer(containerId);
+                LOG.warn("Container with 'id': '{}' and 'name': '{}' was killed ", containerId, containerName);
+            }
+        } catch (IOException e) {
+            LOG.error("Failed to kill unused container with 'id': '{}' and 'name': '{}'", containerId, containerName, e);
```

Comment:
In that case exception will be converted to string. You have to use String.format here.